### PR TITLE
[RLlib] Aggregate impala learner info

### PR DESCRIPTION
**Pull request by avnishn**
Aggregate learner infos for impala training step fn

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ArturNiederfahrenhorst — APPROVED** [review]

File: rllib/algorithms/impala/impala.py [hunk 1]
```diff
@@ -865,7 +873,7 @@ def process_experiences_tree_aggregation(
 
     def update_workers_if_necessary(self) -> None:
         # Only need to update workers if there are remote workers.
-        global_vars = {"timestep": self._counters[NUM_AGENT_STEPS_SAMPLED]}
+        global_vars = {"timestep": self._counters[NUM_AGENT_STEPS_TRAINED]}
```
Comment (sven1977) [inline rllib/algorithms/impala/impala.py:876]: Just out of curiosity: These should be the same, unless I use the mixin buffer with ratio > 0.0, correct?
I feel like we don't really have any opinionated rule of when to use sampled steps and when to use trained steps. IMO, we should always count both.

File: rllib/algorithms/impala/impala.py [hunk 2]
```diff
@@ -795,7 +795,8 @@ def place_processed_samples_on_learner_queue(self) -> None:
 
     def process_trained_results(self) -> ResultDict:
         # Get learner outputs/stats from output queue.
```
Comment (sven1977) [inline rllib/algorithms/impala/impala.py:797]: The comment is outdated here. Could you explain via some more one-line comments what we do here?

**Review by sven1977 — COMMENTED** [review]
Looks great! thanks for the fix @avnishn . Could you just add a few more one-line comments explaining why we sometimes need to deepcopy the threads infos (when there is nothing to process ...)?

File: rllib/algorithms/impala/impala.py (hunk 1 above)
Comment (avnishn) [reply rllib/algorithms/impala/impala.py:876]: Hmmmmm im not sure if this is the case -- We definitely can sample more timesteps than we have trained on at any given moment. 

In practice it doesn't make sense to adjust learning rates and entropy coefficients until the agent has been updated, hence why I changed it here from sampled to trained.

**Review by ArturNiederfahrenhorst — APPROVED** [review]

File: rllib/algorithms/impala/impala.py (hunk 2 above)
Comment (avnishn) [inline rllib/algorithms/impala/impala.py:797]:
```suggestion
        # Combine learner stats from the learner queue and update relevant timestep counters
```

**Pull request merged**